SOC-3238 | fix: founder three tips, check if discussions exist

mleczey opened the pull request:
The days passed emails are not properly showing discussion links.  Ticket here:

https://wikia-inc.atlassian.net/browse/SOC-3238
## Reviewers

@garthwebb

mleczey committed:
SOC | fix: founder three tips, check if discussions exist

WikiaTech commented:
SonarQube analysis reported 26 issues
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L70](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L70): Define a constant instead of duplicating this literal "salutation" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L71](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L71): Define a constant instead of duplicating this literal "editorProfilePage" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L72](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L72): Define a constant instead of duplicating this literal "editorUserName" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L73](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L73): Define a constant instead of duplicating this literal "editorAvatarURL" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L74](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L74): Define a constant instead of duplicating this literal "summary" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L75](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L75): Define a constant instead of duplicating this literal "buttonText" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L76](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L76): Define a constant instead of duplicating this literal "buttonLink" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L77](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L77): Define a constant instead of duplicating this literal "contentFooterMessages" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L82](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L82): Define a constant instead of duplicating this literal "details" 16 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L83](https://github.com/Wikia/app/blob/fc44188bc6732d371c3c4b84f29ef32b949c0ddf/extensions/wikia/Email/Controller/FounderController.class.php#L83): Define a constant instead of duplicating this literal "hasContentFooterMessages" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

Commit:
SOC-3238 Added a way to test founder days passed emails.

These are run via cron and under the same city ID.  Here is the command as run by cron with the extra --test-days argument added and required --wikiId parameter added:

SERVER_ID=177 php extensions/wikia/FounderEmails/maintenance.php --event=daysPassed --test-days=3 --wikiId=125951

WikiaTech commented on extensions/wikia/Email/EmailController.class.php:
```diff
@@ -507,15 +507,13 @@ protected function getUserFromName( $username ) {
+			// Allow an anonymous user to be specified
+			return \User::newFromId( 0 );
 		} else {
 			$user = \User::newFromName( $username, $validate = false );
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$validate" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

WikiaTech commented:
SonarQube analysis reported 53 issues
- ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 7 major
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 42 minor
- ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [EmailController.class.php#L8](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/EmailController.class.php#L8): Class "EmailController" has 49 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [EmailController.class.php#L171](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/EmailController.class.php#L171): Remove this unused "$contentType" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
3. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [EmailController.class.php#L173](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/EmailController.class.php#L173): Remove this unused "$priority" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
4. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [EmailController.class.php#L174](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/EmailController.class.php#L174): Remove this unused "$attachments" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
5. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [EmailController.class.php#L175](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/EmailController.class.php#L175): Remove this unused "$sourceType" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
6. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [FounderEmailsEvent.class.php#L85](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/FounderEmails/FounderEmailsEvent.class.php#L85): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
7. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L70](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/Controller/FounderController.class.php#L70): Define a constant instead of duplicating this literal "salutation" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
8. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L71](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/Controller/FounderController.class.php#L71): Define a constant instead of duplicating this literal "editorProfilePage" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
9. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L72](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/Controller/FounderController.class.php#L72): Define a constant instead of duplicating this literal "editorUserName" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
10. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [FounderController.class.php#L73](https://github.com/Wikia/app/blob/ee2866604e05c571738c9dfefb0c506429a77898/extensions/wikia/Email/Controller/FounderController.class.php#L73): Define a constant instead of duplicating this literal "editorAvatarURL" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/FounderEmails/maintenance.php:
```diff
@@ -39,6 +41,16 @@
 $wikiId = null;
 if ( isset( $options['wikiId'] ) && is_numeric( $options['wikiId'] ) ) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "wikiId" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

garthwebb commented on extensions/wikia/Email/EmailController.class.php:
```diff
@@ -507,15 +507,13 @@ protected function getUserFromName( $username ) {
 			throw new Fatal( 'Required username has been left empty' );
 		}
 
-		// Allow an anonymous user to be specified
```
Moved this below the test for `$username` being an object.  When `$username` is an object this test emits a warning about  being unable to convert the object into an integer.

garthwebb commented on extensions/wikia/FounderEmails/FounderEmailsEvent.class.php:
```diff
@@ -109,16 +109,17 @@ public static function register() {
 	}
 
-	public function create() {
+	public function create( $wikiId = null ) {
```
Allow a wiki ID to be passed in so that we can create an event for any wiki.  Used with the --test-days parameter of the maintenance script changed in this PR.

garthwebb commented on extensions/wikia/FounderEmails/maintenance.php:
```diff
@@ -19,9 +19,11 @@
 		--event			specific email event i.e. daysPassed, viewsDigest, completeDigest
 		--wikiId		specific wiki id
+		--test-days     test daysPassed email with a specific day value.  Creates fake record that 
+		                will always trigger.  Requires --wikiId.
```
Using `--test-days` allows us to test the daysPassed email.  A wiki ID is always required because we need to create a fake row for a specific wiki.

garthwebb commented:
LGTM
